config/m4/opt.m4

```diff
@@ -1,23 +1,36 @@
 # opt.m4 - Macros to control optimization
 # 
-# Copyright (C) Mellanox Technologies Ltd. 2016.  ALL RIGHTS RESERVED.
+# Copyright (C) Mellanox Technologies Ltd. 2016-2017.  ALL RIGHTS RESERVED.
 # See file LICENSE for terms.
 #
 
 ##########################
 # Logging control
 #
+# VMA_OPTIMIZE_LOG values:
+# 5 - VMA_DEBUG and up
+# 6 - VMA_FINE and up
+#
 AC_ARG_ENABLE(
     [opt-log],
     AC_HELP_STRING(
         [--enable-opt-log],
-        [Optimize logging output (default=no)]))
+        [Optimize logging output: none, medium, high (default=medium)]),,
+    enableval=medium)
 AC_MSG_CHECKING(
     [checking for logging optimization])
-if test "x$enable_opt_log" = "xyes"; then
-    CPPFLAGS="$CPPFLAGS -DVMA_OPTIMIZE_LOG -DNDEBUG"
-    AC_MSG_RESULT([yes])
-else
-    AC_MSG_RESULT([no])
-fi
+case "$enableval" in
+    yes | medium)
+        CPPFLAGS="$CPPFLAGS -DVMA_OPTIMIZE_LOG=6 -DNDEBUG"
+        ;;
+    no | none)
```

Comment:
I am suggesting `VMA_OPTIMIZE_LOG=10` i.e. to use value that exceeds maximum level.